### PR TITLE
[NEW] Redesign quoted messages

### PR DESCRIPTION
**Pull request by gerzonc**
## Proposed changes

Changes quoted messages design to match with how it looks on the Web client and fixes attachments on quoted messages. Applies to: images, videos, audios and plain text.
## Issue(s)	

Closes #3396
Closes #2564 

#### Simple quote
1. Send a message
2. Reply to that message

Should render the message and the quoted message properly.

#### Nested quote
1. Send a message
2. Reply to that message
3. Reply to the previous message

Should render all the messages, included the most nested one.

#### Image message
1. Send a message with an image
2. Reply to that message

Should render the message and the quoted message with image.

#### Video message
1. Send a message with a video
2. Reply to that message

Should render the message and the quoted message with video.

## Screenshots
<table>
  <thead>
    <tr>
      <th>Before</th>
      <th>After</th>
      </thead>
    </tr>
    <tbody>
      <tr>
        <td>
        image
        </td>
        <td>
        image
        </td>
      </tr>
    </tbody>
</table >

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by diegolmello — CHANGES_REQUESTED** [review]
A couple of questions:
- We should apply use real name
- Quote time is misaligned
- Did we run Detox?
- Should we play the audio on quote or just redirect?
- Add spacing between quote and message, like on Figma

**Comment by gerzonc** [comment]
> * We should apply use real name?
We don't receive that information on the request, just `author_name` which give us the username.

> * Did we run Detox?
Not really 🙈

> * Should we play the audio on quote or just redirect?
I think we should just redirect... I thought about playing the audio, but UX for jumping to the original message won't leave enough area for you to jump to it besides the corners.

Other questions were not responded as I assume they are just part of the review, will work on those changes.

**Comment by ankar84** [comment]
Nice to have that improvement in 4.26!

**Review by diegolmello — APPROVED** [review]
LGTM

File: app/containers/message/Attachments.tsx [hunk 1]
```diff
@@ -29,42 +35,55 @@ const AttachedActions = ({ attachment, theme }: IMessageAttachedActions) => {
...
+	({ attachments, timeFormat, showAttachment, style, getCustomEmoji, isReply }: IMessageAttachments) => {
 		if (!attachments || attachments.length === 0) {
 			return null;
 		}
 
-		return attachments.map((file: any, index: number) => {
-			if (file.image_url) {
+		if (attachments.length === 2 && attachments[0].image_url && attachments[1].message_link) {
+			attachments = [attachments[1], attachments[0]];
+		}
```
Comment (gerzonc) [inline app/containers/message/Attachments.tsx:45]: This looks ugly but is the only way I could think of in order to achieve `Image` to be after `Reply` 🙈 
image

File: app/containers/message/Audio.tsx [hunk 2]
```diff
@@ -287,12 +287,11 @@ class MessageAudio extends React.Component<IMessageAudioProps, IMessageAudioStat
...
 						animateTransitions
 						animationConfig={sliderAnimationConfig}
-						thumbTintColor={isAndroid && themes[theme].tintColor}
+						thumbTintColor={isReply ? themes[theme].tintDisabled : isAndroid && themes[theme].tintColor}
 						minimumTrackTintColor={themes[theme].tintColor}
 						maximumTrackTintColor={themes[theme].auxiliaryText}
 						onValueChange={this.onValueChange}
-						/* @ts-ignore*/
-						thumbImage={isIOS && { uri: 'audio_thumb', scale }}
+						thumbImage={isIOS ? { uri: 'audio_thumb', scale } : undefined}
```
Comment (gerzonc) [inline app/containers/message/Audio.tsx:294]: Can't we just add the `audio_thumb` asset to Android? I think it's better so it looks the same in both platforms

**Pull request merged**